src/diagnostics/MOM_wave_speed.F90

```diff
@@ -81,7 +81,7 @@ subroutine wave_speed(h, tv, G, GV, US, cg1, CS, full_halos, use_ebt_mode, mono_
                                           !! for the purposes of calculating vertical modal structure [nondim].
   real,                   optional, intent(in)  :: mono_N2_depth !< A depth below which N2 is limited as
                                           !! monotonic for the purposes of calculating vertical
-                                          !! modal structure [Z ~> m].
+                                          !! modal structure [H ~> m or kg m-2].
```

Comment:
should `Hf`, `Hc`, `htot`, `hmin`, `H_here` also be `[H ~> m or kg m-2]`

Reply:
All of these will be converted to thickness units of `[H ~> m or kg-2]` in the next PR, but I wanted to separate this PR, related to changing the units of these arguments, which requires changes to multiple files, from the primarily internal refactoring in the next PR.